pipeline/pipelines/code_generation_pipeline.py

```diff
@@ -11,97 +11,86 @@
 def _validate_codegen_kwargs(extra_args, **kwargs):
-    required = ['src_proto_path',
-                'import_proto_path',
-                'toolkit_path',
-                'output_dir',
-                'language',
-                'api_name']
+    required = ['src_proto_path', 'import_proto_path', 'toolkit_path',
+                'output_dir', 'language', 'api_name']
     pipeline_util.validate_exists(required + extra_args, **kwargs)
-    _validate_toolkit_path(kwargs['toolkit_path'])
 
 
 def _validate_publish_kwargs(allowed_envs, **kwargs):
-    required = ['repo_url',
-                'username',
-                'password']
+    required = ['repo_url', 'username', 'password']
     if 'publish_env' in kwargs:
         pipeline_util.validate_exists(required, **kwargs)
         if kwargs['publish_env'] not in allowed_envs:
-            raise ValueError(
-                    'Invalid publish_env "{}". choose from one of {}'
-                    .format(kwargs['publish_env'], allowed_envs))
+            raise ValueError('Invalid publish_env "{}". choose from one of {}'
+                             .format(kwargs['publish_env'], allowed_envs))
 
 
 class GapicConfigPipeline(pipeline_base.PipelineBase):
-
     def __init__(self, **kwargs):
         kwargs['language'] = ''
         super(GapicConfigPipeline, self).__init__(**kwargs)
 
     def do_build_flow(self, **kwargs):
         flow = linear_flow.Flow('gapic-configgen')
-        flow.add(protoc_tasks.ProtoDescGenTask('ProtoDesc', inject=kwargs),
-                 gapic_tasks.GapicConfigGenTask(
-                     'GapicConfigGen', inject=kwargs),
-                 gapic_tasks.GapicConfigMoveTask(
-                     'GapicConfigMove', inject=kwargs))
+        flow.add(
+            io_tasks.PrepareGoogleapisDirTask('PrepareGoogleapisDirTask',
+                                              inject=kwargs),
+            protoc_tasks.ProtoDescGenTask('ProtoDesc',
+                                          inject=kwargs),
+            gapic_tasks.GapicConfigGenTask('GapicConfigGen',
+                                           inject=kwargs),
+            gapic_tasks.GapicConfigMoveTask('GapicConfigMove',
+                                            inject=kwargs))
         return flow
 
     def validate_kwargs(self, **kwargs):
         _validate_codegen_kwargs([], **kwargs)
 
 
 class PythonGrpcClientPipeline(pipeline_base.PipelineBase):
-
     def __init__(self, **kwargs):
         kwargs['language'] = 'python'
         super(PythonGrpcClientPipeline, self).__init__(**kwargs)
 
     def do_build_flow(self, **kwargs):
         flow = linear_flow.Flow('grpc-codegen')
-        flow.add(protoc_tasks.GrpcPackmanTask('Packman', inject=kwargs),
-                 package_tasks.GrpcPackageDirTask('PackageDir', inject=kwargs))
+        flow.add(
+            io_tasks.PrepareGoogleapisDirTask('PrepareGoogleapisDirTask',
+                                              inject=kwargs),
```

Comment:
@michaelbausor , please help on the refactoring once this PR gets merged.

@jumk, I will leave it as-is in this PR.